dsa-solutions/lc-solutions/0000-0099/0084-largest-rectangle-in-histogram.md

```diff
@@ -0,0 +1,250 @@
+---
+id: largest-rectangle-in-histogram
+title: Largest Rectangle in Histogram (Leetcode)
+sidebar_label: 0082-LargestRectangleInHistogram
+description: Given an array of integers heights representing the histogram's bar height where the width of each bar is 1, return the area of the largest rectangle in the histogram.
+---
+
+## Problem Description
+
+| Problem Statement | Solution Link | LeetCode Profile |
+| :---------------- | :------------ | :--------------- |
+| [Largest Rectangle in Histogram ](https://leetcode.com/problems/largest-rectangle-in-histogram/description/) | [Largest Rectangle in Histogram Solution on LeetCode](https://leetcode.com/problems/largest-rectangle-in-histogram/solutions) |  [Aaradhya Singh ](https://leetcode.com/u/keira_09/) |
+
+
+## Problem Description
+
+Given an array of integers heights representing the histogram's bar height where the width of each bar is 1, return the area of the largest rectangle in the histogram.
+
+### Examples
+
+#### Example 1
+
+- **Input:** $heights = [2,1,5,6,2,3]$
+- **Output:** $10$
+- **Explanation:** The above is a histogram where width of each bar is 1.
+The largest rectangle is shown in the red area, which has an $area = 10$ units.
+
+
+#### Example 2
+
+- **Input:** $heights = [2,4]$
+- **Output:** $4$
+
+### Constraints
+
+
+- 1 <= heights.length <= 105

```

Comment:
**Bad practice**

```md
- 1 <= heights.length <= 105
- 0 <= heights[i] <= 104
```

**Good Practice**

```md
- $1 <= heights.length <= 105$
- $0 <= heights[i] <= 104$
```

Reply:
Alright... I'm new to this setup.. I'll resolve the issue ASAP.